### ISSUE TITLE
`image` misparsed

### ISSUE THREAD

**Issue by mbutterick** [greghendershott/markdown#80]
```
![image](test.jpg)
```

is parsed as

```
'((div ((class "figure")) (img ((src "test.jpg") (alt "image"))) (p ((class "caption")) "image")))
```

But this variant:

```
foo ![image](test.jpg) bar
```

parses quite differently:

```
'((p () "foo " (img ((src "test.jpg") (alt "image"))) " bar"))
```

Whereas the [majority of implementations](https://babelmark.github.io/?text=!%5Bimage%5D(test.jpg)) — assign whatever credibility you like to their opinion — parse the first one this way:

```
'(p (img ((src "test.jpg")(alt "image"))))
```

And the [second like so](https://babelmark.github.io/?text=foo+!%5Bimage%5D(test.jpg)+bar):

```
'(p "foo " (img ((src "test.jpg")(alt "image"))) " bar")
```

So `markdown` agrees with the majority in the second case, but not in the first case. Because `markdown` seems to be inconsistent with itself, I would say the first case is a bug.

**Comment by greghendershott** [comment]
I agree they should be consistent --- for the value of the [`current-strict-markdown?`](https://docs.racket-lang.org/markdown/index.html#(def._((lib._markdown%2Fparse..rkt)._current-strict-markdown~3f))) parameter.

- When `current-strict-markdown?` is true (the default), I think both should parse to the common/"plain" thing you see in other markdown parsers.

- When `current-strict-markdown?` is false, I do like the "fancy" parse with the caption.

Also Frog uses `current-struct-markdown?` false. I think I'd rather have the second thing start parsing "more fancy" --- as opposed to the first thing start parsing "less fancy". I think that would be the less-bad change?

What do you think?

**Comment by greghendershott** [comment]
p.s. Looking at [the doc](https://docs.racket-lang.org/markdown/index.html#(def._((lib._markdown%2Fparse..rkt)._current-strict-markdown~3f))), today, I have to admit that another valid issue to open would be: "Provide useful information so people don't need to dive into the source code to learn what features this toggles."

**Comment by mbutterick** [comment]
Ah I see — I didn’t know about `current-strict-markdown?` So maybe there isn’t any bug, because this:

```racket
#lang racket
(require markdown)
(current-strict-markdown? #true)
(parse-markdown "![image](test.jpg)")
```

Does in fact produce the hoped-for parse:

``` 
'((p () (img ((src "test.jpg") (alt "image")))))
```

Still, the docs are wrong about this:

```racket
#lang racket
(require markdown)
(current-strict-markdown?) ; #false by default, but docs promise #true
```

So either the docs or the default value should change, to be consistent with one another, I think.

**Comment by greghendershott** [comment]
I pushed a commit with a draft of improved documentation.

Not only does it now correctly say the default is `#t`, it itemizes what functionality this parameter controls.

**Comment by greghendershott** [comment]
I haven't merged to `master` if you want to take a look and give feedback. Otherwise I'll merge within a couple days.

**Comment by mbutterick** [comment]
Nothing further, thanks.

**Comment by greghendershott** [comment]
I merged commit a312c20 to `master`. The docs should refresh within 24 hours.
